From d07f5baa6326dde08d5a462a68a248083081e316 Mon Sep 17 00:00:00 2001
From: Rod Whitby <rod@whitby.id.au>
Date: Sun, 27 Jan 2008 10:36:52 +1030
Subject: [PATCH] ixp4xx: Register dsmg600 rtc i2c_board_info

Register the i2c board info related to the RTC chip on the dsmg600
board to allow it to be found automatically on boot.

Signed-off-by: Rod Whitby <rod@whitby.id.au>
Signed-off-by: Alessandro Zummo <a.zummo@towertech.it>
---
 arch/arm/mach-ixp4xx/dsmg600-setup.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-ixp4xx/dsmg600-setup.c b/arch/arm/mach-ixp4xx/dsmg600-setup.c
index 1e75e10..14bc67a 100644
--- a/arch/arm/mach-ixp4xx/dsmg600-setup.c
+++ b/arch/arm/mach-ixp4xx/dsmg600-setup.c
@@ -14,6 +14,7 @@
 #include <linux/kernel.h>
 #include <linux/serial.h>
 #include <linux/serial_8250.h>
+#include <linux/i2c.h>
 
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
@@ -48,6 +49,12 @@ static struct platform_device dsmg600_i2c_controller = {
 	.dev.platform_data	= &dsmg600_i2c_gpio_pins,
 };
 
+static struct i2c_board_info __initdata dsmg600_i2c_board_info [] = {
+	{
+		I2C_BOARD_INFO("rtc-pcf8563", 0x51),
+	},
+};
+
 #ifdef CONFIG_LEDS_CLASS
 static struct resource dsmg600_led_resources[] = {
 	{
@@ -155,6 +162,9 @@ static void __init dsmg600_init(void)
 
 	pm_power_off = dsmg600_power_off;
 
+	i2c_register_board_info(0, dsmg600_i2c_board_info,
+				ARRAY_SIZE(dsmg600_i2c_board_info));
+
 	/* The UART is required on the DSM-G600 (Redboot cannot use the
 	 * NIC) -- do it here so that it does *not* get removed if
 	 * platform_add_devices fails!
-- 
1.5.2.5

